### PR TITLE
Adds docker setup for BastionAI server

### PR DESCRIPTION
**Pull request by kbamponsem**
Adds docker build file to the server.
- Currently, the docker build is made for `libtorch-cpu`.

### CODE REVIEW

File: client/bastionai/client.py [hunk 1]
```diff
@@ -211,6 +211,7 @@ class Connection:
     server_name: str = 'bastionai-srv'
 
     def __enter__(self) -> Client:
+
```
Comment (dhalf) [inline client/bastionai/client.py:214]: Remove blank line (use black)
It'll be easier once the CI works

File: client/bastionai/utils.py [hunk 2]
```diff
@@ -117,8 +117,6 @@ def dataset_from_chunks(chunks: Iterator[Chunk]) -> TensorDataset:
     if any([x is None for x in columns]):
         raise Exception(f"Missing column in data wrapper.")
 
-    return TensorDataset(columns, labels)
```
Comment (dhalf) [inline client/bastionai/utils.py:120]: Why do you remove this return statement?????

File: server/Makefile [hunk 3]
```diff
@@ -1,6 +1,6 @@
-SERVER_DIR:=./
+SERVER_DIR := $(shell pwd)
 BIN_DIR := $(SERVER_DIR)/bin
-SERVER_NAME := server
+SERVER_NAME := bastionai_app
```
Comment (dhalf) [inline server/Makefile:3]: Will these changes affect the way we build the server currently?

File: server/bastionai_app/Cargo.toml [hunk 4]
```diff
@@ -18,7 +18,7 @@ tokio-stream = "0.1"
 # http = "0.2.8"
 # anyhow = "1.0.58"
 ring = "0.16.20"
-private_learning = { path = "../../private_learning" }
+private_learning = { path = "../private_learning" }
```
Comment (dhalf) [inline server/bastionai_app/Cargo.toml:21]: This is likely incompatible with PR#35

File: client/bastionai/utils.py (hunk 2 above)
Comment (kbamponsem) [reply client/bastionai/utils.py:120]: It wasn't removed. I branched when the new API wasn't made.

File: server/Makefile (hunk 3 above)
Comment (kbamponsem) [reply server/Makefile:3]: Not much.
We just have to rename the package in the Cargo.toml file

File: server/bastionai_app/Cargo.toml (hunk 4 above)
Comment (kbamponsem) [reply server/bastionai_app/Cargo.toml:21]: Yeah.

File: client/bastionai/utils.py (hunk 2 above)
Comment (dhalf) [reply client/bastionai/utils.py:120]: Oh I see, so this is PR will require a merge from master at some point

File: server/Makefile (hunk 3 above)
Comment (dhalf) [reply server/Makefile:3]: Sounds good

**Pull request merged**